lib/metadata/include/irods/metadata.hpp

```diff
@@ -0,0 +1,163 @@
+#ifndef IRODS_METADATA_HPP
+#define IRODS_METADATA_HPP
+
+#include <string>
+#include <vector>
+#include <map>
+
+#include "irods/entity.hpp"
+#include "irods/rcConnect.h"
+
+#ifdef IRODS_METADATA_ENABLE_SERVER_SIDE_API
+    #include "irods/rsModAVUMetadata.hpp"
+    using rxComm_t = rsComm_t;
+    #define rxModAVUMetadata rsModAVUMetadata
+    #define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
+    #define IRODS_QUERY_ENABLE_SERVER_SIDE_API
+#else
+    #include "irods/modAVUMetadata.h"
+    using rxComm_t = rcComm_t;
+    #define rxModAVUMetadata rcModAVUMetadata
+#endif
+
+#include "irods/irods_query.hpp"
+
+
+namespace irods::experimental::metadata {
+
+          using entity_type = irods::experimental::entity::entity_type;
+          struct avu {
+                  std::string attribute;
+                  std::string value;
+                  std::string units;
+              };
+        bool operator==(const avu& _lhs, const avu& _rhs) noexcept;
+
+namespace {
```

Comment:
This namespace and everything inside of it should be moved to the cpp file.

These things look like implementation details. The developer doesn't need to see them.